src/GraphQL.NewtonsoftJson/GraphQLBuilderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+#nullable enable
+
+using System;
+using GraphQL.DI;
+using Newtonsoft.Json;
+
+namespace GraphQL.NewtonsoftJson
+{
+    /// <inheritdoc cref="GraphQL.GraphQLBuilderExtensions"/>
+    public static class GraphQLBuilderExtensions
+    {
+        /// <summary>
+        /// Registers the Newtonsoft.Json <see cref="DocumentWriter"/> as a singleton of type
+        /// <see cref="IDocumentWriter"/> within the dependency injection framework and configures
+        /// it with the specified configuration delegate.
+        /// </summary>
+        public static IGraphQLBuilder AddNewtonsoftJson(this IGraphQLBuilder builder, Action<JsonSerializerSettings>? action = null)
+            => builder.AddDocumentWriter<DocumentWriter>().Configure(action);
+
+        /// <inheritdoc cref="AddNewtonsoftJson(IGraphQLBuilder, Action{JsonSerializerSettings})"/>
+        public static IGraphQLBuilder AddNewtonsoftJson(this IGraphQLBuilder builder, Action<JsonSerializerSettings, IServiceProvider>? action)
```

Comment:
`IServiceProvider>? action` -> `IServiceProvider> action`

Reply:
We could rewrite all of these _secondary_ overloads to have a non-nullable delegate as you indicated.  I probably wouldn't, for greater flexibility when creating 3rd party extensions that rely on these overloads.  Marking them as nullable does not impede any user code.  But if you feel strongly about it, then that's fine.

I would make my decision without considering if it helps the `.AddMemoryCache(null)` example.  The scenario is a figment of imagination and does not matter to users.

Reply:
OK, keep as is.